docs/development.md

```diff
@@ -55,3 +55,131 @@ Pre-commit should now work as expected when making commits even without the need
+!!! note
+
+    Messages sent via Kafka are referred to as "broadcast messages" in the drunc
+    codebase and documentation.
+
+Kafka is known for being complex to set up and manage. The Docker Compose stack provides
+a Kafka image packaged by Bitnami which seems to have worked reliably so far.
+Configuration for the Kafka service is carried out using environment variables in
+`docker-compose.yml`. The configuration values used are not well understood so it is not
+recommended to change them unless systemic issues are encountered.
+
+Longer term the DUNE project may move away from Kafka to a more lightweight solution.
+The web application should aim to be as agnostic as possible to the underlying messaging
+system.
+
+### drunc
+
+This service is only used as part of the `drunc` Docker Compose profile. See the [Docker
+Setup overview] for details.
+
+Runs the various components of the DUNE control system that the web app interfaces with.
+Drunc has a complex set of dependencies collectively referred to the as the DUNE DAQ
+environment (see the [DUNE DAQ documentation] for more information). The upshot is that
+it is not feasible for us to build our own Drunc image from scratch for development.
+Instead we use a base image published by the Dune project that provides the required
+dependencies.
+
+The base images are published by the [daq-release repository]. There are separate images
+for nightly ([nightly-release-alma9]) and release ([frozen-release-alma9]) builds. We
+are currently tracking nightly builds as we are dependent on cutting edge drunc
+developments. It may be feasible in the future to swap to a pinned release build.
+
+To ensure a consistent developer experience we pin the development image to a fixed base
+image. The base image is manually tagged and pushed to the
+[imperialcollegelondon/dunedaq_dev_environment] package with a tag including the date of
+the nightly.
+
+The primary function of the drunc image is to run the Drunc Process Manager which is
+then responsible for booting the processes of a Drunc session. The Process Manager
+launches processes on localhost via SSH. To support this the `drunc` service also starts
+an SSH server with keys configured to allow passwordless connection to localhost.
+
+#### Updating the drunc base image
+
+The following changes are needed to update the version of the base image used for the
+`drunc` service.
+
+1. Pull the latest nightly image - `docker pull
+   ghcr.io/dune-daq/nightly-release-alma9:development_v5`.
+1. Retag the image for the imperialcollegelondon/dunedaq_dev_environment package using
+   the date of the new nightly, e.g.
+   `docker tag ghcr.io/dune-daq/nightly-release-alma9:development_v5
+   ghcr.io/imperialcollegelondon/dunedaq_dev_environment:nightly-241114`.
+1. Update `drunc_docker_service/Dockerfile` to the reflect the date of the new nightly.
+   You'll need to change the `NIGHTLY_TAG` environment variable and the name of the base
+   image.
```

Comment:
Maybe mention that only the date part of the `NIGHTLY_TAG` should be updated, so `NFD_DEV_241114_A9` becomes `NFD_DEV_NEWDATE_A9`, eg. `NFD_DEV_241206_A9`.

Reply:
Done